From 584699e6d5be55bbaa01c04346c81786fdd847f7 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Tue, 3 Apr 2018 15:27:50 +0800
Subject: [PATCH 25/25] arm64: dts: add dts for updating FPGA configure

Add a overlay to implement updating FPGA congifure in Linux
enviroment in real time via Stratix 10 FPGA manager.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/altera/Makefile                     |  2 +-
 .../boot/dts/altera/socfpga_stratix10_fpga_update.dts   | 17 +++++++++++++++++
 2 files changed, 18 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/altera/socfpga_stratix10_fpga_update.dts

diff --git a/arch/arm64/boot/dts/altera/Makefile b/arch/arm64/boot/dts/altera/Makefile
index 5b08e85..ecc0560 100644
--- a/arch/arm64/boot/dts/altera/Makefile
+++ b/arch/arm64/boot/dts/altera/Makefile
@@ -1,4 +1,4 @@
-dtb-$(CONFIG_ARCH_STRATIX10) += socfpga_stratix10_socdk.dtb
+dtb-$(CONFIG_ARCH_STRATIX10) += socfpga_stratix10_socdk.dtb socfpga_stratix10_fpga_update.dtb
 dtb-$(CONFIG_ARCH_STRATIX10SWVP) += stratix10_swvp.dtb
 
 always		:= $(dtb-y)
diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10_fpga_update.dts b/arch/arm64/boot/dts/altera/socfpga_stratix10_fpga_update.dts
new file mode 100644
index 0000000..c7811cc
--- /dev/null
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10_fpga_update.dts
@@ -0,0 +1,17 @@
+/dts-v1/;
+/plugin/;
+/ {
+	fragment@0 {
+		target-path = "/soc/base_fpga_region";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			firmware-name = "soc_s10_fpga_config.rbf";
+			config-complete-timeout-us = <2000000>;
+		};
+	};
+};
+
-- 
2.8.1

